packages/destination-actions/src/destinations/engage-messaging-sendgrid/sendEmail/index.ts

```diff
@@ -288,7 +302,13 @@ const action: ActionDefinition<Settings, Payload> = {
       return
     } else if (['subscribed', 'true'].includes(emailProfile?.subscriptionStatus)) {
       statsClient?.incr('actions-personas-messaging-sendgrid.subscribed', 1, tags)
-      const traits = await fetchProfileTraits(request, settings, payload.userId, statsClient, tags)
+
+      let traits
+      if (payload.traitEnrichment) {
+        traits = payload?.traits ? payload?.traits : JSON.parse('{}')
+      } else {
+        traits = await fetchProfileTraits(request, settings, payload.userId, statsClient, tags)
+      }
```

Comment:
```suggestion
      let traits
      if (payload.traitEnrichment) {
        traits = payload?.traits ?? {}
      } else {
        traits = await fetchProfileTraits(request, settings, payload.userId, statsClient, tags)
      }
```

I think you might be able to simplify this but you may need to change the Profile interface on like 110.

Reply:
yeah we thought about that but since we are keeping the profile api call behind the gate for now, it did not make sense to change the interface for our use case. 

btw, is `payload?.traits ? payload?.traits : JSON.parse('{}')` and `payload?.traits ?? {}` same thing.

Reply:
Yeah at least that's what I was hoping. Just less verbose.

Reply:
it complains with that change 
`(property) Profile.traits: Record<string, string>
Type '{ [k: string]: unknown; }' is not assignable to type 'Record<string, string>'.
  'string' index signatures are incompatible.
    Type 'unknown' is not assignable to type 'string'.ts(2322)
index.ts(113, 3): The expected type comes from property 'traits' which is declared here on type 'Profile'`

I recall that this was the reason why I had to do the JSON.parse thing.

Reply:
If typescript is going to fight us we don't need to solve it in this PR.